pkg/cmd/config/config.go

```diff
@@ -0,0 +1,180 @@
+		Progress: "plain",
+	}
+}
+
+// Set a configuration key and value.
+func setConfig(_ context.Context, opts *configOptions) error {
+	configPath := getConfigPath(opts.profile)
+	cfg, err := LoadConfig(configPath)
+	if err != nil {
+		cfg = DefaultConfig()
+	}
+
+	v := reflect.ValueOf(cfg).Elem().FieldByName(strings.Title(opts.key))
+	if !v.IsValid() {
+		return fmt.Errorf("unknown configuration key: %s", opts.key)
+	}
+
+	v.SetString(opts.value)
+	err = SaveConfig(cfg, configPath) // Save only when configuration is modified.
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Config '%s' set to '%s'\n", opts.key, opts.value)
+	return nil
+}
+
+// Get a configuration value.
+func getConfig(_ context.Context, opts *configOptions) (string, error) {
+	configPath := getConfigPath(opts.profile)
+	cfg, err := LoadConfig(configPath)
+	if err != nil {
+		return "", err
+	}
+
+	v := reflect.ValueOf(cfg).Elem().FieldByName(strings.Title(opts.key))
+	if !v.IsValid() {
+		return "", fmt.Errorf("unknown configuration key: %s", opts.key)
+	}
+
+	return fmt.Sprintf("%v", v.Interface()), nil
+}
+
+// List all configuration values.
+func listConfig(_ context.Context, opts *configOptions) error {
+	configPath := getConfigPath(opts.profile)
+	cfg, err := LoadConfig(configPath)
+	if err != nil {
+		return err
+	}
+
+	// Use reflection to iterate through fields and print them.
+	v := reflect.ValueOf(cfg).Elem()
+	t := v.Type()
+	for i := 0; i < v.NumField(); i++ {
+		fmt.Printf("%s: %v\n", t.Field(i).Name, v.Field(i).Interface())
+	}
+	return nil
+}
+
+// Reset configuration to defaults.
+func resetConfig(_ context.Context, opts *configOptions) error {
+	configPath := getConfigPath(opts.profile)
+	cfg := DefaultConfig()
+	err := SaveConfig(cfg, configPath)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Configuration reset to default values.")
+	return nil
+}
+
+// Load configuration from a file.
+func LoadConfig(configPath string) (*Config, error) {
+	if configPath == "" {
+		return nil, fmt.Errorf("config path is empty")
+	}
+
+	file, err := os.Open(configPath)
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			// Return default config, but don't save it to the file.
+			return DefaultConfig(), nil
+		}
+		return nil, fmt.Errorf("error opening config file: %w", err)
+	}
+	defer file.Close()
+
+	var config Config
+	if err := json.NewDecoder(file).Decode(&config); err != nil {
+		return nil, fmt.Errorf("error decoding config file: %w", err)
+	}
+
+	// If some fields are empty, fallback to defaults.
+	defaultConfig := DefaultConfig()
+	if config.LogLevel == "" {
+		config.LogLevel = defaultConfig.LogLevel
+	}
+	if config.Progress == "" {
+		config.Progress = defaultConfig.Progress
+	}
+
+	return &config, nil
+}
+
+// Save configuration to a file.
+func SaveConfig(config *Config, configPath string) error {
+	// Ensure the directory exists before saving the file.
+	dir := filepath.Dir(configPath)
+	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
```

Comment:
Restrict permissions to the owner only.

```suggestion
	if err := os.MkdirAll(dir, 0700); err != nil {
```